fcrepo-import/src/main/java/org/fcrepo/importer/Importer.java

```
@@ -17,26 +17,178 @@
+ *
  * @author awoods
+ * @author escowles
  * @since 2016-08-29
  */
 public class Importer implements TransferProcess {
+    private static final Logger logger = getLogger(Importer.class);
+    private Config config;
+    protected FcrepoClient client;
 
     /**
      * Constructor that takes the Import/Export configuration
      *
      * @param config for import
      */
     public Importer(final Config config) {
+        this.config = config;
+        this.client = FcrepoClient.client().build();
     }
 
     /**
      * This method does the import
      */
     public void run() {
         System.out.println("Importing!");
+        importDirectory(config.getDescriptionDirectory());
+    }
+
+    private void importDirectory(final File dir) {
+        for (final File f : dir.listFiles()) {
+            if (f.isDirectory()) {
+                importDirectory(f);
+            } else if (f.isFile()) {
+                importFile(f);
+            }
+        }
+    }
+
+    private void importFile(final File f) {
+        FcrepoResponse response = null;
+        URI uri = null;
+        try {
+            final Model model = createDefaultModel().read(new FileInputStream(f), null,
+                    config.getRdfLanguage());
+            final ResIterator binaryResources = model.listResourcesWithProperty(RDF_TYPE, NON_RDF_SOURCE);
+            if (binaryResources.hasNext()) {
+                uri = new URI(binaryResources.nextResource().getURI());
+                logger.info("Importing binary {}", f.getAbsolutePath());
+                response = importBinary(uri, sanitize(model));
+            } else {
+                uri = uriForFile(f, config.getDescriptionDirectory());
+                logger.info("Importing container {}", f.getAbsolutePath());
+                response = importContainer(uri, sanitize(model));
+            }
+
+            if (response.getStatusCode() > 204) {
+                logger.warn("Error while importing {} ({}): {}",
```

We should probably throw an exception here, no?